modules/vision/src/pose-estimation/vpPoseDementhon.cpp

```diff
@@ -466,9 +469,34 @@ void vpPose::poseDementhonPlan(vpHomogeneousMatrix &cMo)
   }
   vpColVector sv;
   vpMatrix Ap, imA, imAt, kAt;
-  int irank = A.pseudoInverse(Ap, sv, 1.e-6, imA, imAt, kAt);
-  if (irank != 3) {
-    throw(vpException(vpException::fatalError, "In Dementhon planar, rank is not 3"));
+  bool isRankEqualTo3(false); // Indicates if the rank of A is the expected one
+  double logNofSvdThresh = std::log(dementhonSvThresh)/lnOfSvdFactorUsed; // Get the log_n(dementhonSvThresh), where n is the factor by which we will multiply it if the svd decomposition fails.
+  int nbMaxIter = std::max(std::ceil(logNOfSvdThresholdLimit - logNofSvdThresh), 1.); // Ensure that if the user chose a threshold > svdThresholdLimit, at least 1 iteration of svd decomposition is performed
+  double svdThreshold = dementhonSvThresh;
+  int irank(0);
+  for(int i = 0; i < nbMaxIter && !isRankEqualTo3; i++)
+  {
+    irank = A.pseudoInverse(Ap, sv, svdThreshold, imA, imAt, kAt);
+    if(irank == 3)
+    {
+      isRankEqualTo3 = true;
+    }
+    else
+    {
+      isRankEqualTo3 = false;
+      svdThreshold *= svdFactorUsedWhenFailure;
+    }
+  }
+  
+  if (!isRankEqualTo3) {
+    std::string errorMsg("In Dementhon planar, after ");
+    errorMsg += std::to_string(nbMaxIter);
+    errorMsg += std::string(" trials multiplying the svd threshold by ");
+    errorMsg += std::to_string(nbMaxIter);
+    errorMsg += std::string(", rank (");
+    errorMsg += std::to_string(irank);
+    errorMsg += std::string(") is still not 3");
+    throw(vpException(vpException::fatalError, errorMsg));
```

Comment:
Same here. `std::to_string` is c++11.
With c++98 we should use `std::stringstream`.

In the next release we will keep c++98 compat. It will be the last release compatible with  c++98.

Reply:
Again, did not think about being c++98 compatible.
Changed in 279c1e7acb7e675682c481644b95263774853f0a , lines 492 to 500